go/service/user.go

```diff
@@ -344,6 +345,179 @@ func (h *UserHandler) UploadUserAvatar(ctx context.Context, arg keybase1.UploadU
 	return avatars.UploadImage(mctx, arg.Filename, nil /* teamname */, arg.Crop)
 }
 
+func (h *UserHandler) ProofSuggestions(ctx context.Context, sessionID int) (ret keybase1.ProofSuggestionsRes, err error) {
+	mctx := libkb.NewMetaContext(ctx, h.G()).WithLogTag("US")
+	defer mctx.CTraceTimed("ProofSuggestions", func() error { return err })()
+	suggestions, err := h.proofSuggestionsHelper(mctx)
+	if err != nil {
+		return ret, err
+	}
+	foldPriority := mctx.G().GetProofServices().SuggestionFoldPriority()
+	for _, suggestion := range suggestions {
+		if foldPriority > 0 && suggestion.Priority >= foldPriority {
+			ret.ShowMore = true
+			suggestion.BelowFold = true
+		}
+		ret.Suggestions = append(ret.Suggestions, suggestion.ProofSuggestion)
+	}
+	return ret, nil
+}
+
+type ProofSuggestion struct {
+	keybase1.ProofSuggestion
+	Priority int
+}
+
+var dummyIcon []keybase1.SizedImage // TODO CORE-9882: Get some icons
+
+func (h *UserHandler) proofSuggestionsHelper(mctx libkb.MetaContext) (ret []ProofSuggestion, err error) {
+	user, err := libkb.LoadMe(libkb.NewLoadUserArgWithMetaContext(mctx).WithPublicKeyOptional())
+	if err != nil {
+		return ret, err
+	}
+	if user == nil {
+		return ret, fmt.Errorf("could not load logged-in user")
+	}
+
+	var suggestions []ProofSuggestion
+	serviceKeys := mctx.G().GetProofServices().ListServicesThatAcceptNewProofs()
+	for _, service := range serviceKeys {
+		switch service {
+		case "web", "dns", "http", "https":
+			// These are under the "web" umbrella.
+			// "web" is added below.
+			continue
+		}
+		serviceType := mctx.G().GetProofServices().GetServiceType(service)
+		if serviceType == nil {
+			mctx.CDebugf("missing proof service type: %v", service)
+			continue
+		}
+		if len(user.IDTable().GetActiveProofsFor(serviceType)) > 0 {
+			mctx.CDebugf("user has an active proof: %v", serviceType.Key())
+			continue
+		}
+		subtext := serviceType.DisplayGroup()
+		if len(subtext) == 0 {
+			subtext = serviceType.PickerSubtext()
+		}
+		suggestions = append(suggestions, ProofSuggestion{ProofSuggestion: keybase1.ProofSuggestion{
+			Key:           service,
+			ProfileText:   fmt.Sprintf("Prove your %v", serviceType.DisplayName()),
+			ProfileIcon:   dummyIcon,
+			PickerText:    serviceType.DisplayName(),
+			PickerSubtext: subtext,
+			PickerIcon:    dummyIcon,
+		}})
+	}
+	hasPGP := len(user.GetActivePGPKeys(true)) > 0
+	if !hasPGP {
+		suggestions = append(suggestions, ProofSuggestion{ProofSuggestion: keybase1.ProofSuggestion{
+			Key:           "pgp",
```

Comment:
can these suggestions be constants defined outside of the function so it reads easier:
`suggestions = append(suggestions, PGPProofSuggestion)`